api4/permission.go

```diff
@@ -0,0 +1,30 @@
+package api4
+
+import (
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+func (api *API) InitPermissions() {
+	api.BaseRoutes.Permissions.Handle("/ancillary_permissions", api.ApiSessionRequired(getAncillaryPermissions)).Methods("GET")
+}
+
+func getAncillaryPermissions(c *Context, w http.ResponseWriter, r *http.Request) {
+	keys, ok := r.URL.Query()["subsection_permissions"]
+
+	if !ok || len(keys[0]) < 1 {
```

Comment:
Do we need to validate that `subsection_permissions` contains *only* subsection permissions? What happens if you pass it other permissions?

Reply:
Nothing , it will just return the permissions you passed it. So if I do subsection_permissions=notpermission1, notpermission2

We will just get back [notpermission1, notpermission2]